opmd_viewer/openpmd_timeseries/interactive.py

```diff
@@ -116,7 +116,27 @@ def refresh_ptcl(force=False) :
                         species=ptcl_species_button.value, plot=True,
                         vmin=vmin, vmax=vmax, cmap=ptcl_color_button.value,
                         nbins=ptcl_bins_button.value )
-                
+
+        def refresh_species(b):
+            """
+            Refresh the particle species buttons
+            """
+            # Get available records for this species (remove charge and
+            # mass as they are typically less interesting)
+            avail_records = [ q for q in \
+                        self.avail_record_components[ptcl_species_button.value]
+                        if (q in ['charge', 'mass'])==False ]
+            # Update the plotting buttons
+            ptcl_xaxis_button.options = avail_records
+            ptcl_xaxis_button.value = avail_records[0]
+            ptcl_yaxis_button.options = avail_records + ['None']
+            ptcl_yaxis_button.value = 'None'
+            # Update the selection widgets
+            for dropdown_button in ptcl_select_widget.quantity:
+                dropdown_button.options = avail_records
+            
+            refresh_ptcl()
+                    
         def refresh_ptcl_now(b):
```

Comment:
note for future improvements: you can use [`functools.partial`](https://docs.python.org/2/library/functools.html#functools.partial) for that job: it binds default arguments to functions so you can use them in callbacks with the right signature

```python
from functools import partial

refresh_ptcl_now = partial(refresh_ptcl, True)
ptcl_refresh_button.on_click( refresh_ptcl_now )
```

(C++ term: binding)